src/testRunner/unittests/tsserver/projectErrors.ts

```diff
@@ -833,3 +834,29 @@ describe("unittests:: tsserver:: projectErrors:: with file rename on wsl2::", ()
         baselineTsserverLogs("projectErrors", "file rename on wsl2", session);
     });
 });
+
+describe("unittests:: tsserver:: projectErrors:: dts errors when files dont belong to common root", () => {
+    [undefined, "decls"].forEach(declarationDir => {
+        it(`dts errors when files dont belong to common root${declarationDir ? " with declarationDir" : ""}`, () => {
+            const host = createServerHost({
+                "/home/src/projects/project/src/file.ts": `import { a } from "../a";`,
+                "/home/src/projects/project/a.ts": `export const a = 10;`,
+                "/home/src/projects/project/src/tsconfig.json": jsonToReadableText({
+                    compilerOptions: {
+                        composite: true,
+                        noEmit: true,

```

Comment:
We use to calculate common source dir and report error if `OutDir`, `rootDir`, `mapRoot` or `sourceRoot` were specified or files would be emitted. So if you had `noEmit` we wouldnt cache this and then when getting dts diagnostics it would crash because it was over eager (or didnt cache for `declarationDir`